components/admin_real/desktop/ReservationFieldsModal.jsx

```diff
@@ -0,0 +1,190 @@
+import React, { useState } from "react";
+import Modal from "antd/lib/modal/Modal";
+import { Alert, DatePicker, Input } from "antd";
+import { TimePicker } from "antd";
+import moment from "moment";
+import { Select } from "antd";
+
+const { Option } = Select;
+
+const ReservationFieldsModal = ({
+  reservation = {},
+  onSubmit,
+  modalTitle,
+  modalOpen = false,
+  setModalOpen,
+}) => {
+  const dateFormat = "YYYY-MM-DD";
+  const timeFormat = "HH:mm";
+  const id = reservation.id || 123;
+
+  const [date, setDate] = useState(reservation.date);
+  const [time, setTime] = useState(reservation.time || 0);
+  const [certified, setCertified] = useState(
+    reservation.diver_certified ? reservation.diver_certified.toString() : null
+  );
+  const [reservationType, setReservationType] = useState(
+    reservation.reservation_type
+  );
+  const [name, setName] = useState(
+    reservation.diver_information ? reservation.diver_information.name : null
+  );
+  const [lastName, setLastName] = useState(
+    reservation.diver_information
+      ? reservation.diver_information.lastName
+      : null
+  );
+  const [age, setAge] = useState(
+    reservation.diver_information ? reservation.diver_information.age : null
+  );
+  const [experience, setExperience] = useState(reservation.number_of_dives);
+  const [email, setEmail] = useState(
+    reservation.diver_information ? reservation.diver_information.email : null
+  );
+
+  const reservationValid = () => {
+    if (
```

Comment:
we can just return all this, instead of the `if/else`

Reply:
That's because you are not casting it into a boolean 😄 , that is how Js works.

i.e. 
```
true && console.log('hello')
```
will console log. The way JS resolves the and operator is, if the first value is a truthy value, it will do whatever the second value is and execute (if its a string, it will do the string, if its a function it will execute a function).

In your case, since `email` is a string, when you return the function, it will return the string.

This is ok with the `if/else`.